Content.Shared/Weapons/Reflect/ReflectComponent.cs

```diff
@@ -0,0 +1,46 @@
+using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.Weapons.Reflect;
+
+/// <summary>
+/// Entities with this component have a chance to reflect projectiles and hitscan shots
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+public sealed class ReflectComponent : Component
+{
+    /// <summary>
+    /// Can only reflect when enabled
+    /// </summary>
+    [DataField("enabled"), ViewVariables(VVAccess.ReadWrite)]
+    public bool Enabled = true;
+
+    [DataField("hitscanChance"), ViewVariables(VVAccess.ReadWrite)]
+    public float HitscanChance;
+
+    [DataField("projectileChance"), ViewVariables(VVAccess.ReadWrite)]
+    public float ProjectileChance;
```

Comment:
These should be the same; hitscan / projectile should be interchangeable (e.g. if someone wants a hitscan ballistic weapon, or disablers which are projectile weapons which should be similar to lasers). If you want energy to be different it should be via some kind of tag or the likes.